client/files.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// RemovePathOptions holds the options for a call to RemovePath.
+type RemovePathOptions struct {
+	// Path is the absolute path to be deleted (required).
+	Path string
+
+	// Recursive, if true, will delete all files and directories contained
+	// within the specified path, recursively. Defaults to false.
+	Recursive bool
+}
+
+type removePathsPayload struct {
+	Action string            `json:"action"`
+	Paths  []removePathsItem `json:"paths"`
+}
+
+type removePathsItem struct {
+	Path      string `json:"path"`
+	Recursive bool   `json:"recursive"`
+}
+
+type fileResult struct {
+	Path  string       `json:"path"`
+	Error *errorResult `json:"error,omitempty"`
+}
+
+type errorResult struct {
```

Comment:
Thinking about this now, we should be able to JSON-decode this directly to `*client.Error`, which already has these fields and the JSON tags. Similar for the other PRs.